lib/OpenQA/Schema/ResultSet/Jobs.pm

```diff
@@ -317,9 +317,12 @@ sub prepare_complex_query_search_args ($self, $args) {
             push(@conds, {'me.id' => {-in => $subquery->get_column('job_id')->as_query}});
         }
 
-        for my $key (qw(build distri version flavor arch test machine)) {
+        for my $key (qw(distri version flavor arch test machine)) {
             push(@conds, {"me." . uc($key) => $args->{$key}}) if $args->{$key};
         }
+        if (my $build = $args->{build}) {
+            push @conds, ref $build eq 'ARRAY' ? {"me.BUILD" => {-in => $build}} : {"me.BUILD" => $build};
```

Comment:
I would not duplicate the common part and we generally use single quotes when not interpolation is required (despite not all old code has been converted to that style):
```suggestion
            push @conds, {'me.BUILD' => (ref $build eq 'ARRAY' ? {-in => $build} : $build)};
```

Reply:
my suggestion missed that part, sorry

Reply:
Actually not fixed:

```patch
-    push @conds, ref $build eq 'ARRAY' ? {'me.BUILD' => {-in => $build}} : {'me.BUILD' => $build};
+   push @conds, {'me.BUILD' => ref $build eq 'ARRAY' ? {-in => $build} : $build};